app/models.py

```diff
@@ -171,6 +171,28 @@ class Person(SearchableMixin, db.Model):
     education = db.Column(db.String)
     publications = db.Column(db.String)
 
+    def redact_data(self, fields):
+        if not fields["image"]:
+            self.image = None
+        if not fields["email"]:
+            self.email = None
+        if not fields["room"]:
+            self.entryway = None
+            self.floor = None
+            self.suite = None
+            self.room = None
+        if not fields["phone"]:
+            self.phone = None
+        if not fields["address"]:
+            self.address = None
+        if not fields["major"]:
+            self.major = None
+        if not fields["birthday"]:
+            self.birthday = None
+            self.birth_month = None
+            self.birth_day = None
```

Comment:
I'm fine with this implementation, but I also wonder if it could be made a little less repetitive by having something like a dictionary of things that can be removed, along with additional properties to remove with them:
```python
RELATED_FIELDS = {
'image': [],
'email': [],
'room': ['entryway', 'floor', 'suite'],
'phone': [],
'address': [],
'major': [],
'birthday': ['birth_day', 'birth_month'],
}
```
And then you could just iterate through and use getattr/setattr to do what you gotta do without so much repetition. But it's up to you as that could also make the code a bit less readable.

Also nitpick but swap the quotes

Reply:
Quotes fixed. I'm personally partial to the if ladder, even though it is a little big, because it's only `room` and `birthday` that have multiple fields, adding a total of like 4 lines.